stdpopsim/cli.py

```diff
@@ -625,22 +694,49 @@ def run_simulation(args):
             f"Running simulation model {model.id} for {species.id} on "
             f"{contig} with {len(samples)} samples using {engine.id}."
         )
-        dfe_interval = args.dfe_interval
-        dfe = args.dfe
-        if dfe_interval is None:
-            dfe_interval = ",".join(
-                map(str, [0, int(contig.recombination_map.sequence_length)])
-            )
-        if dfe is not None:
+        # DFE assignment
+        dfe = None
+        dfe_summary_str = None
+        if args.dfe is None:
+            if args.dfe_interval is not None:
+                exit(
+                    "A DFE interval has been assigned without a DFE. "
+                    "Please specify a DFE "
+                )
+            elif args.dfe_annotation is not None:
+                exit(
+                    "A DFE annotation has been assigned without a DFE. "
+                    "Please specify a DFE "
+                )
+        else:
+            if args.dfe_interval is not None:
+                if args.dfe_annotation is not None:
+                    exit(
+                        "A DFE annotation and a DFE interval have been "
+                        "selected. Please only use one "
+                    )
+                left, right = args.dfe_interval.split(",")
+                intervals = np.array([[int(left), int(right)]])
+                dfe_summary_str = f"[{left}, {right})"
+            elif args.dfe_annotation is not None:
+                annot = species.get_annotations(args.dfe_annotation)
+                intervals = annot.get_chromosome_annotations(args.chromosome)
+                dfe_summary_str = f"{annot.id} elements on {args.chromosome}"
+            else:
+                # case where no intervals specified but we have a DFE
+                intervals = np.array(
+                    [[0, int(contig.recombination_map.sequence_length)]]
+                )
+                dfe_summary_str = f"[{intervals[0][0]}, {intervals[0][1]})"
+
             dfe = species.get_dfe(args.dfe)
-            left, right = dfe_interval.split(",")
             contig.add_dfe(
-                intervals=np.array([[int(left), int(right)]]),
+                intervals=intervals,
                 DFE=dfe,
             )
             logger.info(
                 f"Applying selection under the DFE model {dfe.id} "
-                f"in interval [{left}, {right})."
+                f"in intervals {intervals}."
```

Comment:
Isn't this going to print out something absolutely huge when we're using an Annotation? Would it be better to put the summary string here?

Reply:
yep